wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/urlfactory/DefaultFactoryUrl.java

```diff
@@ -61,6 +66,33 @@ public String getBranch() {
     return null;
   }
 
+  @Override
+  public Optional<String> getCredentials() {
+    if (url == null || isNullOrEmpty(url.getUserInfo())) {
+      return Optional.empty();
+    }
+    String userInfo = url.getUserInfo();
+    String[] credentials = userInfo.split(":");
+    String username = credentials[0];
+    String password = credentials.length == 2 ? credentials[1] : null;
+    if (!isNullOrEmpty(username) || !isNullOrEmpty(password)) {
+      return Optional.of(
+          format(
+              "%s:%s",
+              isNullOrEmpty(username) ? "" : username, isNullOrEmpty(password) ? "" : password));
+    }
```

Comment:
I'm a bit lost, can we really handle a case from auth perspective if either username or password is null / blank? I thought we need both, no?

Reply:
Actually we don't need both parts. If we have only username, we pass the credentials as `username:`. From the authentication prospective this case is considered as a token. I agree that it looks a bit weird, but it coms from the `curl` behaviour:
```
curl https://<token>@dev.azure.com/vinokurig/test/_apis/git/repositories/test/items?path=%2Fdevfile.yaml -v                                                                                        
*   Trying 185.199.109.133:443...
* Connected to raw.githubusercontent.com (185.199.109.133) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (OUT), TLS handshake, Client hello (1):
* (304) (IN), TLS handshake, Server hello (2):
* (304) (IN), TLS handshake, Unknown (8):
* (304) (IN), TLS handshake, Certificate (11):
* (304) (IN), TLS handshake, CERT verify (15):
* (304) (IN), TLS handshake, Finished (20):
* (304) (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / AEAD-AES128-GCM-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.io
*  start date: Feb 21 00:00:00 2023 GMT
*  expire date: Mar 20 23:59:59 2024 GMT
*  subjectAltName: host "raw.githubusercontent.com" matched cert's "*.githubusercontent.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS RSA SHA256 2020 CA1
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Server auth using Basic with user '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'
* h2h3 [:method: GET]
* h2h3 [:path: /vinokurig/private/master/devfile.yaml]
* h2h3 [:scheme: https]
* h2h3 [:authority: raw.githubusercontent.com]
* h2h3 [authorization: Basic <token plus : in Base 64> ]
```
So `curl https://<token>@dev.azure.com/...` equals `curl https://dev.azure.com/... -H 'Authorization: Basic <token + : in base64>'`